### ISSUE TITLE
Carriers are not being initialized when editing shipping method

### ISSUE THREAD

**Issue by luzat** [shipcloud/shipcloud-for-woocommerce#200]
When editing shipping methods, there are none shown:
![image](https://user-images.githubusercontent.com/1327084/112837451-32696d00-909c-11eb-8699-452c47446d5d.png)

The problem seems to be one of the following two, not sure which one:

1. `WC_Shipcloud_Shipping::carriers` is only initialized for the settings screen in `init_settings_screen` and thus is not available in `init_settings_fields` to in turn initialize `$carriers_options` for use in `instance_form_fields`' `allowed_carriers` field:
  ```
    	$this->instance_form_fields = array(
			'allowed_carriers'                  => array(
				'title'       => __( 'Shipping methods', 'shipcloud-for-woocommerce' ),
				'type'        => 'multi_checkbox',
				'description' => __( 'Select the carriers that you want to use in your shop.', 'shipcloud-for-woocommerce' ),
				'desc_tip'    => true,
				'options'     => $carriers_options
			),
2. Alternatively or additionally, could it be that `allowed_carriers` should actually be initialized from `available_carriers` here?

Another thing that I noticed is, that `WC_Shipcloud_Shipping::init_allowed_carriers` does not initialize anything. Isn't it meant to at least assign `allowed_carriers` to some instance variable? Instead it seems that most methods seem to do this on their own. It's at least a misnomer.

**Comment by thd-fox** [comment]
Same here! But I face this issue, when initially enter API key on new install.

**Comment by andrecedik** [comment]
Yes I know. We previously had this update upon entering and saving the form. Unfortunately Wordpress always loads the shipping method (multiple times), resulting in multiple calls to the API, which is unnecessary and can get people to reach their request limits. Therefore you currently have to save your settings one more time after entering the API key for them to be loaded.

**Comment by thd-fox** [comment]
Unfortunately this "...save your settings one more time after entering the API key..." doesn't work either.

**Comment by luzat** [comment]
For me, the carriers do display in settings, but not when editing shipping methods, which is because `$this->carriers` is only initialized in this code path:

    if ( wcsc_is_settings_screen() ) {
      if ( ! $this->init_settings_screen() ) {
        return false;
      }
    }

Given that shipping methods are not edited on the shipcloud settings screen, `$this->carriers` is always going to be `[]`.

**Comment by andrecedik** [comment]
Hmm, ok interesting. Will have to have a look at that.
